File: 2021/2021W05/PreppindataW05.py
import pandas as pd

final = pd.read_csv(r'F:/Data/Joined Dataset.csv', parse_dates=['From Date'], date_parser=lambda x: pd.to_datetime(x, format='%d/%m/%Y'))
final['From Date'] = final['From Date'].dt.date
final = final.groupby(['Client ID']).apply(lambda x: x[x['From Date']==x['From Date'].max()]).reset_index(drop=True).copy()

File: 2021/2021W05/preppindataW05.R
library(dplyr)
library(readr)

final <- read_csv("F:/Data/Joined Dataset.csv", 
                  col_types = cols(`From Date` = col_date(format = "%d/%m/%Y"))) %>%
  group_by(`Client ID`) %>%
  filter(`From Date` == max(`From Date`)) %>%
  ungroup()

View(final)
